Set out expectations for `good first issue` label (#14686)
jupyterlab/jupyterlab: docs/source/developer/contributing.rst

Our expectation is that every new issue should be examined within a week of
its creation.

> Triagers are encouraged to label easier issues as ``good first issue`` to
> facilitate beginner contributions; the triage team should ensure that
> sufficient context and links to relevant parts of the codebase are included
> before applying this label.

Reviewer: ```suggestion
Triagers should label easier issues as ``good first issue`` to
facilitate beginner contributions. A good first issue should have:

* A clear, easily understood description, with screen shots and expectations that do not require much familiarity with the product
* Links, either in the description or in comments, to documentation and source code files that are relevant to the issue
* Recommended points of contact, either by GitHub username or on other forums (Discourse, etc) where a developer can get help

Unless an issue is time-sensitive, such as if it is a release blocker
for an imminent release, experienced Jupyter developers should avoid
picking up issues with the ``good first issue`` label.
```

Our expectation is that every new issue should be examined within a week of
its creation.

> Triagers should label easier issues as ``good first issue`` to

Reviewer 2: Suggest replacing "easier" with a less ambiguous term such as low effort/low complexity

Author: That's a bit tricky. I think we should not say that good first issues are low effort - this can be demotivating especially if it ends up being a lot of effort for a beginner; they certainly should not be high complexity issues as there is a compelxity in learning a new codebase and contribution process itself. It might be hard to put this into words. I will use "easier/lower complexity" for now, please suggest a better wording if something comes to mind.

For example, GitHub uses "_selected for its relative approachability_" (https://github.com/jupyterlab/jupyterlab/contribute):

> Each issue displayed here is a "good first issue," selected for its relative approachability for first-time contributors.

That's a mouthful, but maybe good?

facilitate beginner contributions. A good first issue should have:

> * A clear, easily understood description, with screen shots and expectations that do not require much familiarity with the product

Reviewer 2: ```suggestion
* A clear, easily understood description with screen shots and expectations that do not require much familiarity with the project
```

* Links, either in the description or in comments, to documentation and source code files that are relevant to the issue

> * Recommended points of contact, either by GitHub username or on other forums (Discourse, etc) where a developer can get help

Reviewer 2: Suggest replacing developer with contributor
as we often require skills beyond development such as design or accessibility knowledge

Author: Replaced in 29cd2ea.

Unless an issue is time-sensitive, such as if it is a release blocker

> for an imminent release, experienced Jupyter developers should avoid

Reviewer 2: Same suggest using contributors instead

Author: Replaced in 29cd2ea.

> picking up issues with the ``good first issue`` label.

Reviewer 3: Should we add a notion of time? Like old one are good to be tackled by seasoned dev?

```suggestion
picking up recent issues with the ``good first issue`` label.
```
